marlowe-runtime/indexer/Language/Marlowe/Runtime/Indexer/Database/PostgreSQL/GetMarloweUTxO.hs

```diff
@@ -24,49 +24,44 @@ getMarloweUTxO block = runMaybeT do
   slotNo <- MaybeT $ H.statement (prepareBlockQueryParams block) [maybeStatement|
     SELECT slotNo :: bigint
       FROM marlowe.block
-     WHERE id = $1 :: bytea AND slotNo = $2 :: bigint AND blockNo = $3 :: bigint AND rollbackToSlot IS NULL AND rollbackToBlock IS NULL
+     WHERE id = $1 :: bytea AND slotNo = $2 :: bigint AND blockNo = $3 :: bigint
   |]
 
-
   unspentContractOutputs <- lift $ Map.fromDistinctAscList . V.toList . fmap decodeContractOutputRow <$> H.statement slotNo [vectorStatement|
-    WITH block (blockId) AS
```

Comment:
I don't understand why this pattern was problematic, aside from some added complexity.

Reply:
This pattern worked for `marlowe-chain-indexer`. What is the fundamental difference for `marlowe-indexer`?

Reply:
Thanks for the explanation!

Reply:
Thanks for clarifying that "this pattern = having rollback columns in the block table, instead of having a separate rollback table".

I already summarized the problems in my comment below - and as for `marlowe-sync` vs `marlowe-chain-sync`, this problem has also caused bugs in `marlowe-chain-sync`. However, here are the differences:

1. `marlowe-chain-sync` lacks foreign key constraints, so we get fewer guarantees and still need to carefully manage the cascading deletes of other tables when rolling back blocks
2. `marlowe-chain-sync` is currently more mature and mission-critical than `marlowe-sync`, so I am more reluctant to make such a drastic change there

I do however think that a similar change would be an improvement in `marlowe-chain-sync`, I'd just want to approach it a lot more carefully and in a separate PR.